src/context.rs

```diff
@@ -566,26 +576,33 @@ impl<N: FieldExt> Records<N> {
     pub fn assign_range_value(
         &mut self,
         offset: usize,
-        (v, chunks): (N, Vec<N>),
+        (v, decompose_v): (N, Vec<N>),
         leading_bits: u64,
     ) -> AssignedValue<N> {
-        assert!(chunks.len() as u64 <= MAX_CHUNKS);
+        assert!(decompose_v.len() as u64 <= RANGE_VALUE_DECOMPOSE);
         self.ensure_range_record_size(offset + 1 + MAX_CHUNKS as usize);
 
         self.range_fix_record[offset][0] = Some(N::one());
-        self.range_adv_record[offset].0 = Some(v);
+        self.range_adv_record[offset][0].0 = Some(v);
 
         // a row placeholder
         self.range_fix_record[offset + MAX_CHUNKS as usize][0] = Some(N::zero());
 
-        for i in 0..chunks.len() - 1 {
-            self.range_fix_record[offset + 1 + i][1] = Some(N::from(COMMON_RANGE_BITS as u64));
-        }
-        self.range_fix_record[offset + chunks.len()][1] = Some(N::from(leading_bits));
+        for (index, v) in decompose_v.iter().enumerate() {
+            let col = index / MAX_CHUNKS as usize;
+            let row_offset = index % MAX_CHUNKS as usize;
+
+            if col == VALUE_COLUMNS - 1 {
+                self.range_fix_record[offset + 1 + row_offset][1] = if index != decompose_v.len() {
```

Comment:
I guess the index is [0, 1, 2, 3, 4, 5], decompose_v.len() is 6, could index == decompose_v.len()?